src/Models/ApplicantModel.php

```diff
@@ -561,16 +568,54 @@ public function appendToAssessmentNotes(int $id, string $assessmentNote): bool
 
     public function updateEdaid(int $id, int $edaid): bool
     {
-        $query = $this->db->prepare('UPDATE `applicants_additional` SET edaid=:edaid WHERE id=:id');
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET edaid=:edaid WHERE `id` = :id;');
         return $query->execute(["id" => $id, "edaid" => $edaid]);
     }
 
     public function updateGithubUsername(int $id, string $githubUsername): bool
     {
-        $query = $this->db->prepare('UPDATE `applicants_additional` SET githubUsername=:githubUsername
-     WHERE id=:id');
+        $query = $this->db->prepare('UPDATE `applicants_additional` SET githubUsername = :githubUsername
+     WHERE `id` = :id');
         return $query->execute(["id" => $id, "githubUsername" => $githubUsername]);
     }
+    public function toggleLockField(int $id, string $fieldName)
+    {
+        switch ($fieldName) {
+            case 'githubUsername':
+                $locked = $this->isFieldLocked('githubUsernameLocked', $id);
+                $sql = 'UPDATE `applicants_additional` SET `githubUsernameLocked` = :locked WHERE `id` = :id';
+                break;
+            case 'edaid':
+                $locked = $this->isFieldLocked('edaidLocked', $id);
+                $sql = 'UPDATE `applicants_additional` SET `githubUsernameLocked` = :locked WHERE `id` = :id';
+                break;
+            case 'upfront':
+                $locked = $this->isFieldLocked('upfrontLocked', $id);
+                $sql = 'UPDATE `applicants_additional` SET `githubUsernameLocked` = :locked WHERE `id` = :id';
+                break;
+            case 'laptop':
+                $locked = $this->isFieldLocked('laptopLocked', $id);
+                $sql = 'UPDATE `applicants_additional` SET `githubUsernameLocked` = :locked WHERE `id` = :id';
+                break;
+            default:
+                throw new \Exception('Invalid field name');
+        }
+
+        $query = $this->db->prepare($sql);
+        return $query->execute(["id" => $id, "locked" => $locked]);
+    }
+
+    private function isFieldLocked(string $fieldName, int $id): int
+    {
+        $query = $this->db->prepare(
+            'SELECT :fieldName 
```

Comment:
Cannot bind a named placeholder to a field, need to concatenate the field into the query instead. This will create a potential SQL injection vulnerability so will need to mitigate this some how.